js/event.ts

```diff
@@ -0,0 +1,282 @@
+// Copyright 2018 the Deno authors. All rights reserved. MIT license.
+import * as domTypes from "./dom_types";
+
+export const eventAttributes = new WeakMap();
+
+export class EventInit implements domTypes.EventInit {
+  bubbles = false;
+  cancelable = false;
+  composed = false;
+
+  constructor({bubbles=false, cancelable=false, composed=false} = {}) {
+    this.bubbles = bubbles;
+    this.cancelable = cancelable;
+    this.composed = composed;
+  }
+}
+
+export class Event implements domTypes.Event {
+  // Each event has the following associated flags
+  private _stopPropagationFlag = false;
+  private _stopImmediatePropagationFlag = false;
+  private _canceledFlag = false;
+  private _inPassiveListenerFlag = false;
+
+  // Property for objects on which listeners will be invoked
+  private _path: domTypes.EventPath[] = [];
+
+  constructor(type: string, eventInitDict?: domTypes.EventInit) {
+    eventAttributes.set(this, {
+      type,
+      bubbles: eventInitDict && eventInitDict.bubbles || false,
+      cancelable: eventInitDict && eventInitDict.cancelable || false,
+      composed: eventInitDict && eventInitDict.composed || false,
+      currentTarget: null,
+      eventPhase: domTypes.EventPhase.NONE,
+      isTrusted: false,
+      target: null,
+      timeStamp: Date.now(),
+    });
+  }
+
+  get bubbles(): boolean {
+    if (eventAttributes.has(this)) {
+      return eventAttributes.get(this).bubbles || false;
+    }
+
+    throw new TypeError("Illegal invocation");
+  }
+
+  get cancelBubble(): boolean {
+    return this._stopPropagationFlag;
+  }
+
+  get cancelBubbleImmediately(): boolean {
+    return this._stopImmediatePropagationFlag;
+  }
+
+  get cancelable(): boolean {
+    if (eventAttributes.has(this)) {
```

Comment:
Same comments as `bubbles` above... I will stop commenting on this.

Reply:
done